pkg/types/baremetal/validation/platform.go

```diff
@@ -157,6 +158,20 @@ func validateHosts(hosts []*baremetal.Host, fldPath *field.Path) field.ErrorList
 	return hostErrs
 }
 
+func filterHostsBMC(ns []byte) bool {
```

Comment:
Nit: Could we leave a comment here explaining why we validate BMC's separately? Just for future visitors to the code who aren't us.